Code Coverage for qemu_x86 is not getting generated due to a build error

**Describe the bug**
While running the below command for code coverage the tests on qemu_x86 are failing due to a build error. Tests are passing while running sanity check with out the code coverage option. 

~/zephyr/scripts/sanitycheck --coverage --coverage-platform qemu_x86 -p qemu_x86 --coverage-platform qemu_x86_nommu -p qemu_x86_nommu --gcov-tool /opt/zephyr-sdk/i586-zephyr-elf/bin/i586-zephyr-elf-gcov -T ~/zephyr/tests/kernel -T ~/zephyr/tests/subsys/logging | tee output.log

What have you tried to diagnose or workaround this issue?
Build error is observed after the below commit.
Commit ID: fb0faaa1239ad5f6577d3268c67ba168aab57656

**To Reproduce**
Steps to reproduce the behavior:
Run the above command and tests fail due to a build error. 

**Expected behavior**
Tests should pass. 

**Impact**
Blocking the code coverage.

**Screenshots or console output**
[ 83%] Building C object zephyr/CMakeFiles/zephyr_prebuilt.dir/misc/empty_file.c.obj
[ 84%] Linking C executable zephyr_prebuilt.elf
/opt/zephyr-sdk/i586-zephyr-elf/bin/../lib/gcc/i586-zephyr-elf/8.3.0/../../../../i586-zephyr-elf/bin/ld:linker.cmd:161: undefined symbol `_region_min_align' referenced in expression
collect2: error: ld returned 1 exit status
make[3]: *** [zephyr/CMakeFiles/zephyr_prebuilt.dir/build.make:95: zephyr/zephyr_prebuilt.elf] Error 1
make[3]: Target 'zephyr/CMakeFiles/zephyr_prebuilt.dir/build' not remade because of errors.
make[3]: Leaving directory '/home/syerabol/zephyr/sanity-out/qemu_x86/tests/subsys/logging/log_output/logging.log_output'
make[2]: *** [CMakeFiles/Makefile2:480: zephyr/CMakeFiles/zephyr_prebuilt.dir/all] Error 2
make[2]: Target 'all' not remade because of errors.
make[2]: Leaving directory '/home/syerabol/zephyr/sanity-out/qemu_x86/tests/subsys/logging/log_output/logging.log_output'
make[1]: *** [Makefile:84: all] Error 2
make[1]: Target 'default_target' not remade because of errors.
make[1]: Leaving directory '/home/syerabol/zephyr/sanity-out/qemu_x86/tests/subsys/logging/log_output/logging.log_output'

For complete log, please find the attached log.
[build.log](https://github.com/zephyrproject-rtos/zephyr/files/3234507/build.log)

**Environment (please complete the following information):**
 - OS: (e.g. Linux, MacOS, Windows): Fedora
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit SHA or Version fb0faaa1239ad5f6577d3268c67ba168aab57656


**Additional context**
Add any other context about the problem here.

--- a/include/arch/x86/linker.ld
+++ b/include/arch/x86/linker.ld
@@ -196,21 +196,6 @@ SECTIONS
 	/* RAMABLE_REGION */
 	GROUP_START(RAMABLE_REGION)
 
-#ifdef CONFIG_COVERAGE_GCOV
-	SECTION_PROLOGUE(_GCOV_BSS_SECTION_NAME, (NOLOAD),)
-	{
-		MMU_PAGE_ALIGN
-		__gcov_bss_start = .;
-		*(".bss.__gcov0.*");
-		. = ALIGN(4);
-		MMU_PAGE_ALIGN
-		__gcov_bss_end = .;
-	}GROUP_DATA_LINK_IN(RAMABLE_REGION, RAMABLE_REGION)
-
-	__gcov_bss_num_words = ((__gcov_bss_end - __gcov_bss_start) >> 2);
-	__gcov_bss_size = __gcov_bss_end - __gcov_bss_start;
-#endif /* CONFIG_COVERAGE_GCOV */
-
 /* Located in generated directory. This file is populated by the
  * zephyr_linker_sources() Cmake function.
  */

--- a/subsys/testsuite/coverage/coverage_ram.ld
+++ b/subsys/testsuite/coverage/coverage_ram.ld
@@ -6,6 +6,7 @@
 
 /* Copied from linker.ld */
 
+#ifdef CONFIG_ARM
 SECTION_DATA_PROLOGUE(_GCOV_BSS_SECTION_NAME,(NOLOAD),)
 {
 #ifdef CONFIG_USERSPACE
@@ -28,3 +29,19 @@ SECTION_DATA_PROLOGUE(_GCOV_BSS_SECTION_NAME,(NOLOAD),)
 
 __gcov_bss_num_words = ((__gcov_bss_end - __gcov_bss_start) >> 2);
 __gcov_bss_size = __gcov_bss_end - __gcov_bss_start;
+#endif
+
+#ifdef CONFIG_X86
+SECTION_PROLOGUE(_GCOV_BSS_SECTION_NAME, (NOLOAD),)
+{
+	MMU_PAGE_ALIGN
+	__gcov_bss_start = .;
+	*(".bss.__gcov0.*");
+	. = ALIGN(4);
+	MMU_PAGE_ALIGN
+	__gcov_bss_end = .;
+}GROUP_DATA_LINK_IN(RAMABLE_REGION, RAMABLE_REGION)
+
+__gcov_bss_num_words = ((__gcov_bss_end - __gcov_bss_start) >> 2);
+__gcov_bss_size = __gcov_bss_end - __gcov_bss_start;
+#endif

